### PR TITLE
fix(machine): throw error `connection refused` after set proxy

### PR DESCRIPTION
**Pull request by BlackHole1**
When the `machine start` command is executed, Podman automatically retrieves the current host's `*_PROXY` environment variable and assigns it directly to the virtual machine in QEMU. However, most `*_PROXY` variables are set with `127.0.0.1` or `localhost`, such as `127.0.0.1:8888`. This causes failures in network-related operations within the virtual machine due to incorrect proxy settings.

#### Does this PR introduce a user-facing change?

```release-note
Fix a bug proxy IP did not get properly translated when executing `machine start`
```

/cc @Luap99

### CODE REVIEW

File: pkg/machine/qemu/machine_test.go
Comment (BlackHole1) [inline pkg/machine/qemu/machine_test.go:54]: The previous approach could cause unit testing errors in certain situations because it relied on the order of maps, while maps in Golang do not guarantee the order of keys.

**Comment by BlackHole1** [comment]
Hey @vrothberg. Do you mean to modify `Does this PR introduce a user-facing change?`?

**Comment by vrothberg** [comment]
I mean the `commit body message`.  It's very helpful when reviewing a commit and it makes sure that we have a clean git history.  This way, future maintainers have an easier time to understand why certain changes were made.

**Comment by BlackHole1** [comment]
Hey @vrothberg. I have finished making the modifications. Thanks 😄.

File: pkg/machine/qemu/machine_test.go
Comment (Luap99) [reply pkg/machine/qemu/machine_test.go:54]: I think this would benefit from a table based test, i.e. create a slice with all expected strings and then just loop over it:
```go
for _, expect := range variables  {
		assert.Contains(t, string(decodeString), expected)
}
```

Comment (BlackHole1) [reply pkg/machine/qemu/machine_test.go]: Good idea 👍. Modification completed.

**Review by Luap99 — APPROVED** [review]
LGTM

**Review by vrothberg — APPROVED** [review]
/lgtm
/hold

File: pkg/machine/qemu/machine_test.go
Comment (edsantiago) [inline pkg/machine/qemu/machine_test.go:98]: Suggestion:
```go
    for _, item := range tests {
        switch item.expect {
        case "":    // expect it to be unset/empty/missing/FIXME
            FIXME: assert that decodeString does not contain item.key? Or that it contains "item.key=$" ?
        case "=":    // expect it to be exactly what we set it ti
            assert.Contains(t, string(decodeString), fmt.Sprintf("%s=%s", item.key, item.value)
        default:    // expect a transmogrification of what we set it to
            assert.Contains(t, string(decodeString), fmt.Sprintf("%s=%s", item.key, item.expect)
        }
    }
```
And, I don't know what all those '|' pipes are in the original code, if they're a delimiter, maybe it'd be better so `Split()` into an array, then run an array-contains-exact-string assertion instead of the vague `assert.Contains`.

But I really don't know this code, so grain of salt.

**Review by edsantiago — COMMENTED** [review]
There's a lot of duplication, which makes me queasy. And the test for 'contains("")` is anti-useful. See suggestion inline for a possible way to handle `""` (empty), `=` (no need to duplicate duplication), and a default case where you don't need to spell out variables duplicatedly.

Thank you for your initiative and your contribution!

File: pkg/machine/qemu/machine_test.go
Comment (BlackHole1) [reply pkg/machine/qemu/machine_test.go]: Your advice was essential, thanks. I have optimized this code section, and you can now perform a new code review.

File: pkg/machine/qemu/machine_test.go
Comment (edsantiago) [inline pkg/machine/qemu/machine_test.go:91]: Very nice! As a point of general paranoia, this might be safer as `.SplitN(env, "=", 2)` to protect against values containing an equal sign (super unlikely). IMO not worth fixing unless you need to re-push for other reasons; I just mention it as something to think about in the future. Thank you!

File: pkg/machine/ignition.go
Comment (jakecorrenti) [inline pkg/machine/ignition.go:621]: Leftover TODO comment?

**Comment by rhatdan** [comment]
Cleanup the todo in separate PR.
/lgtm
/hold cancel

File: pkg/machine/ignition.go
Comment (BlackHole1) [reply pkg/machine/ignition.go:621]: Don't worry, this will soon be cleaned up in another pull request 🧹

**Pull request merged**
